### PR TITLE
feat(guides): guide hooks

### PR DESCRIPTION
**Pull request by scefali**
This PR adds a hook for guides when they update so that we can turn on/off external guide systems. See related PR: https://github.com/getsentry/getsentry/pull/7353

### CODE REVIEW

File: static/app/types/hooks.tsx [hunk 1]
```diff
@@ -203,6 +210,13 @@ export type SettingsHooks = {
   'settings:organization-navigation-config': SettingsConfigHook;
 };
 
+/**
+ * Callback hooks
```
Comment (evanpurkhiser) [inline static/app/types/hooks.tsx:214]: Can you expand on the description a little bit actually

Comment (scefali) [reply static/app/types/hooks.tsx:211]: @evanpurkhiser Specifically for the `callback:on-guide-update` hook?

Comment (evanpurkhiser) [reply static/app/types/hooks.tsx:211]: nah, what this callback hooks category might be fore basically

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/stores/guideStore.tsx [hunk 2]
```diff
@@ -225,7 +226,7 @@ const storeConfig: GuideStoreDefinition = {
    *  - If the user has already seen the guide, don't show the guide
    *  - Otherwise show the guide
    */
-  updateCurrentGuide() {
+  updateCurrentGuide(dismissed?: boolean) {
```
Comment (Neo-Zhixing) [inline static/app/stores/guideStore.tsx:229]: When is this called with dismissed = true?

Comment (scefali) [reply static/app/stores/guideStore.tsx:229]: @Neo-Zhixing when the user dismisses a guide

**Pull request merged**